### PR TITLE
Unify the typing of the agents' reply function and update their docstrings.

### PR DESCRIPTION
**Pull request by DavdGao**
## Description

1. Unify the typing of the agents' reply function as follows to provide more clear and correct guidance in typing.
```
    def reply(self, x: Optional[Union[Msg, Sequence[Msg]]] = None) -> Msg:
```

2. Update the docstrings of the reply function. 

## Checklist

Please check the following items before code is ready to be reviewed.

- [x]  Code has passed all tests
- [x]  Docstrings have been added/updated in Google Style
- [x]  Documentation has been updated
- [x]  Code is ready for review

### CODE REVIEW

**Review by xieyxclack — COMMENTED** [review]
LGTM

File: src/agentscope/agents/text_to_image_agent.py [hunk 1]
```diff
@@ -51,12 +58,16 @@ def reply(self, x: dict = None) -> dict:
                 x = self.memory.get_memory()[-1]
             else:
                 # if no message find, just return None
```
Comment (qbc2016) [inline src/agentscope/agents/text_to_image_agent.py:60]: Please modify this comment accordingly.

Comment (DavdGao) [reply src/agentscope/agents/text_to_image_agent.py]: Done

**Review by pan-x-c — APPROVED** [review]
LGTM

**Pull request merged**